web/src/main/java/no/nav/modiapersonoversikt/legacy/sak/service/saf/SafService.kt

```diff
@@ -45,109 +45,100 @@ class SafServiceImpl : SafService {
-        val jsonQuery = dokumentoversiktBrukerJsonQuery(fnr)
+    companion object {
+        private val SAF_GRAPHQL_BASEURL: String = EnvironmentUtils.getRequiredProperty("SAF_GRAPHQL_URL")
+        private val SAF_HENTDOKUMENT_BASEURL: String = EnvironmentUtils.getRequiredProperty("SAF_HENTDOKUMENT_URL")
 
-        val response = client.newCall(
-            veilederAutorisertClient(SAF_GRAPHQL_BASEURL)
-                .post(RequestBody.create(jsonType, jsonQuery))
-                .build()
-        ).execute()
+        @KtorExperimentalAPI
+        private val graphQLClient = LoggingGraphqlClient("SAF", URL(SAF_GRAPHQL_BASEURL))
+
+        const val VEDLEGG_START_INDEX = 1
 
-        return handterStatus(response)
+        val JOURNALPOSTTYPE_INN = Journalposttype.I
+        val JOURNALPOSTTYPE_UT = Journalposttype.U
+        val JOURNALPOSTTYPE_INTERN = Journalposttype.N
     }
 
-    override fun hentDokument(journalpostId: String, dokumentInfoId: String, variantFormat: Dokument.Variantformat): TjenesteResultatWrapper {
-        val url = lagHentDokumentURL(journalpostId, dokumentInfoId, variantFormat)
+    override fun hentJournalposter(fnr: String): ResultatWrapper<List<DokumentMetadata>> {
+        val variables = HentBrukersDokumenter.Variables(
+            HentBrukersDokumenter.BrukerIdInput(
+                id = fnr,
+                type = HentBrukersDokumenter.BrukerIdType.FNR
+            )
+        )
 
-        val response = client.newCall(
-            veilederAutorisertClient(url).build()
-        ).execute()
-        return when (response.code()) {
-            200 -> TjenesteResultatWrapper(response.body()?.bytes())
-            else -> handterDokumentFeilKoder(response.code())
+        return runBlocking {
+            val response = HentBrukersDokumenter(graphQLClient).execute(variables, userTokenAuthorizationHeaders)
+            if (response.errors.isNullOrEmpty()) {
+                val data = requireNotNull(response.data)
+                    .dokumentoversiktBruker
+                    .journalposter
+                    .filterNotNull()
+                    .map { DokumentMetadata().fraSafJournalpost(it) }
+                ResultatWrapper(data, emptySet())
+            } else {
+                ResultatWrapper(emptyList(), setOf(Baksystem.SAF))
+            }
         }
     }
 
     override fun hentSaker(ident: String): GraphQLResponse<Hentbrukerssaker.Result> {
-        TODO("Not yet implemented")
+        val variables = if (ident.length == 11) {
+            Hentbrukerssaker.Variables(
+                Hentbrukerssaker.BrukerIdInput(
+                    id = ident,
+                    type = Hentbrukerssaker.BrukerIdType.FNR
+                )
+            )
+        } else {
+            Hentbrukerssaker.Variables(
+                Hentbrukerssaker.BrukerIdInput(
+                    id = ident,
+                    type = Hentbrukerssaker.BrukerIdType.AKTOERID
+                )
+            )
+        }
```

Comment:
Dette er kanskje litt pirkete, men jeg synes kanskje det blir mer lesbart med noe à la sånn her, for nå måtte jeg legge inn en innsats for å lete etter forskjellene mellom hva som sto i de to blokkene 🤔 
```
val identType = if (ident.length == 11) Hentbrukerssaker.BrukerIdType.FNR else Hentbrukerssaker.BrukerIdType.AKTOERID
val variables = Hentbrukerssaker.Variables(
    Hentbrukerssaker.BrukerIdInput(
        id = ident,
        type = identType
    )
)
```